### ISSUE TITLE
No longer use PHP for "New & Noteworthy"

### ISSUE THREAD

**Issue by BeckerWdf** [eclipse-platform/www.eclipse.org-eclipse#122]
Eclipse Foundation proposes to retire PHP. See: https://gitlab.eclipse.org/eclipsefdn/helpdesk/-/issues/1934 and
the following mail on the eclipse.org-committers mailing list:

> Greetings folks,
> 
> If your Eclipse project is using PHP for its website on [https://eclipse.dev/](https://eur03.safelinks.protection.outlook.com/?url=https%3A%2F%2Feclipse.dev%2F&data=05%7C02%7Cma.becker%40sap.com%7C72cc56102a7e48428b9408dc28eca6ee%7C42f7676cf455423c82f6dc2d99791af7%7C0%7C0%7C638430244087038509%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C0%7C%7C%7C&sdata=oVcP280Kgcvg16vFsNF2j2%2BcN%2FwIq4JC2bc12MZJA2s%3D&reserved=0) or for downloads, please be aware that we are proposing its deprecation.
> 
> It may have made sense to allow PHP project content on our servers in the past, but in today's context, it does not. Most projects now support a static website built by a CI mechanism. The same also applies to download pages, where PHP was typically used to read the content of the filesystem in real time, this can be done at build/promotion time.
> 
> Please see [https://gitlab.eclipse.org/eclipsefdn/helpdesk/-/issues/1934](https://eur03.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgitlab.eclipse.org%2Feclipsefdn%2Fhelpdesk%2F-%2Fissues%2F1934&data=05%7C02%7Cma.becker%40sap.com%7C72cc56102a7e48428b9408dc28eca6ee%7C42f7676cf455423c82f6dc2d99791af7%7C0%7C0%7C638430244087049003%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C0%7C%7C%7C&sdata=YuPb3LagIYyjAEtWryG35l7VDo%2B2Jdi48C1Tir%2BXQls%3D&reserved=0) for the details. The short:
> 
> - We are proposing to remove PHP capabilities for project content on [https://eclipse.dev/](https://eur03.safelinks.protection.outlook.com/?url=https%3A%2F%2Feclipse.dev%2F&data=05%7C02%7Cma.becker%40sap.com%7C72cc56102a7e48428b9408dc28eca6ee%7C42f7676cf455423c82f6dc2d99791af7%7C0%7C0%7C638430244087057199%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C0%7C%7C%7C&sdata=tBHHbHA1SgSy6gc8H2y9UvPkqNElhsJWE4STZBa3mSY%3D&reserved=0) by Dec 31, 2024.
> 
> - We are proposing to remove PHP capabilities for project content on DOWNLOADS and ARCHIVES by Dec 31, 2025.
> 
> 
> 
> [https://gitlab.eclipse.org/eclipsefdn/helpdesk/-/wikis/Eclipse-Service/Infrastructure-changes-Guide](https://eur03.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgitlab.eclipse.org%2Feclipsefdn%2Fhelpdesk%2F-%2Fwikis%2FEclipse-Service%2FInfrastructure-changes-Guide&data=05%7C02%7Cma.becker%40sap.com%7C72cc56102a7e48428b9408dc28eca6ee%7C42f7676cf455423c82f6dc2d99791af7%7C0%7C0%7C638430244087063554%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C0%7C%7C%7C&sdata=tB9JlqkIlGkWGoKURg139SN8y7xH%2BpmUepgrpTJ5hoI%3D&reserved=0)
> 
> 
> 
> Do note that, in today's security context, removing PHP is unavoidable. We realize that the EF has many items in a deprecation state - either in progress, or as is the case here, proposed and in planning. This is the unfortunate consequence of maintaining an infrastructure that has not properly evolved as the environment around us has changed, and even in yesteryear's security context, are no longer fit or are simply unacceptable.
> 
> Thank you for reading. Please comment on Issue 1934 (link above).

The New & Noteworthy document uses some PHP. E.g. see: https://eclipse.dev/eclipse/news/4.31/platform.php

I even don't know what these PHP scripts do. Can't we simply produce static HTML files instead?

**Comment by jukzi** [comment]
I guess everybody would be willing to write new stuff in markdown. The main question to me is if the old N&Ns need to be converted - which probably does not make fun.
For me it would be fine if old N&N would still be in the repository but not populated to the webpage anymore.

**Comment by merks** [comment]
It’s mostly already html with a little bit of php wrapper.  Perhaps some JavaScript magic could be used instead to render it.

**Comment by BeckerWdf** [comment]
> It’s mostly already html with a little bit of php wrapper. Perhaps some JavaScript magic could be used instead to render it.

Can you explain what this wrapper does and why it's needed?

**Comment by merks** [comment]
@BeckerWdf 

Just compare this page

https://eclipse.dev/eclipse/news/4.31/platform.html

to this page:

https://eclipse.dev/eclipse/news/4.31/platform.php

which just does this wrapping to create the common theming/scaffolding:

![image](https://github.com/eclipse-platform/www.eclipse.org-eclipse/assets/208716/ea0d6536-6a8b-4c2e-91eb-0bfd4d06ba2d)

There must be another way to do that without statically copying the entire scaffolding.  We might ask the Foundation experts about that...

**Comment by merks** [comment]
> I agree that this must be possible. Should I ask in the linked gitlab-ticket?

Yes, let's ask there.  It's of course in the Foundation's best interest too that we are able to present a common, modern, look-and-feel...

**Comment by BeckerWdf** [comment]
> > I agree that this must be possible. Should I ask in the linked gitlab-ticket?
> 
> Yes, let's ask there. It's of course in the Foundation's best interest too that we are able to present a common, modern, look-and-feel...

I asked there: https://gitlab.eclipse.org/eclipsefdn/helpdesk/-/issues/1934#note_1754553

**Comment by BeckerWdf** [comment]
The Acknowledgement page also uses PHP

**Comment by subyssurendran666** [comment]
I just came across this discussion and have some questions regarding it.

- What will happen to PHP web pages after deprecation takes effect? Will PHP content still be displayed in the browser without the interpreter?
- Has anyone started converting the dynamic content to static content?
- I recently checked the Eclipse download pages, and the majority of the content is already available in static form (both PHP and HTML files are available). We could release the download pages with HTML content. The only drawback is that the static files do not include the header, footer, menu, breadcrumbs, etc.

**Comment by MohananRahul** [comment]
if so, Also we need to change some link php file to html , before RC2 to reflect html page. eg:
https://github.com/eclipse-platform/eclipse.platform.releng.aggregator/blob/235b668774a722043d5dbc12a075c9b280441780/eclipse.platform.releng.tychoeclipsebuilder/eclipse/publishingFiles/templateFiles/index.template_master.php#L106

**Comment by gireeshpunathil** [comment]
I suggest the following:

 - replace all php files with html files
 - replace all php links with html links
 - render the pages and see if it is readable
 - commit the changes
 - tolerate the look & feel issues as of now
 - incrementally fix the LAF over time
 
this way, we address the security issues on priority, while being pragmatic.
